lib/CBattleCallback.cpp

```diff
@@ -1020,8 +1020,28 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo &info) c
 		multBonus *= 0.5;
 	}
 
-	// TODO attack on petrified unit 50%
 	// blinded unit retaliates
+	auto blinded = info.attackerBonuses->getBonus(Selector::source(Bonus::SPELL_EFFECT, SpellID::BLIND));
+	if(blinded)
+	{
+		switch(blinded->val)
+		{
+			case SecSkillLevel::BASIC:

```

Comment:
Air shield gives GENERAL_DAMAGE_REDUCTION bonus.
This piece of code effectively does nothing and is outdated.

On the other hand, the code indicates that distance penalty and air shield penalty should not be cumulative. Strategija reads:
"Attacker is a shooter with range, wall or melee penalty, or Advanced Air Shield is cast"

Though I'm not sure it's correct or logical.